lib/vm.js

```diff
@@ -85,14 +85,41 @@ function validateString(prop, propName) {
                                'string', prop);
 }
 
+function validateBool(prop, propName) {
+  if (prop !== undefined && typeof prop !== 'boolean')
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', propName,
+                               'boolean', prop);
+}
+
+function validateObject(prop, propName) {
+  if (prop !== undefined && (prop === null || typeof prop !== 'object'))
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', propName,
+                               'object', prop);
+  return true;
+}
+
 function getContextOptions(options) {
   if (options) {
     const contextOptions = {
       name: options.contextName,
-      origin: options.contextOrigin
+      origin: options.contextOrigin,
+      codeGeneration:
+        validateObject(options.contextCodeGeneration,
+                       'options.contextCodeGeneration') ? {
```

Comment:
The logic doesn’t currently work when `options.contextCodeGeneration` is undefined.